sdk/search/test-resources.json

```diff
@@ -55,9 +55,16 @@
               "description": "The url suffix to use when accessing the storage data plane."
           }
         },
+        "enableCmk": {
+            "type": "bool",
+            "defaultValue": false,
+            "metadata": {
+                "description": "Whether to enable deployment of Managed CMK. The default is false."
+            }
+        },
         "searchSku": {
             "type": "string",
-            "defaultValue": "basic",
+            "defaultValue": "[if(equals(parameters('enableCmk'), bool('true')), 'basic', 'free')]",
```

Comment:
@heaths [Weekly tests](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=2043470&view=results) are working fine but Nightly pipeline is failing at [Deploy test resources](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=2044091&view=logs&j=ff3a09b3-827d-5654-2747-8202a4b0e892&t=0c3c1616-d3f4-570e-5882-d7fd6c8ea151) step with `Operation would exceed 'free' tier service quota. You are using 1 out of 1 'free' tier service quota` error.

Seems like we need to use the "basic" tier for Nightly runs as well where we don't run Key Vault + CMK tests. I'm not sure how the tests were running with "free" tier before [this](https://github.com/Azure/azure-sdk-for-net/pull/16484) PR in Nov 2020 because [here](https://github.com/MicrosoftDocs/azure-docs/blob/main/includes/azure-search-limits-per-subscription.md) I don't see Search RP has changed it's limit since then. I'm not sure what changed and there's no great way to verify anymore.

Reply:
Does running the CMK tests all the time and deploying a keyvault cost us anything significant, especially if we have to use a "basic" SKU anyway? do we need to introduce all these complexity, if we can't use "free" SKU?

I see other languages are also using "basic" tier" - [Java](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/search/test-resources.json#L35), [JS](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/search/test-resources.json#L14)